src/core/Entry.cpp

```diff
@@ -609,12 +609,12 @@ void Entry::truncateHistory()
             // don't calculate size if it's already above the maximum
             if (size <= histMaxSize) {
                 size += historyItem->attributes()->attributesSize();
-
-                const QSet<QByteArray> newAttachments = historyItem->attachments()->values().toSet() - foundAttachments;
-                for (const QByteArray& attachment : newAttachments) {
-                    size += attachment.size();
+                size += historyItem->autoTypeAssociations()->associationsSize();
+                size += historyItem->attachments()->attachmentsSize(foundAttachments);
+                foreach( const QString &tag, historyItem->tags().split(QRegExp(",|;|:"), QString::SkipEmptyParts)){
```

Comment:
Can you split this line in 2?  It isn't very clear

Also, can you use C++11 range `for` instead of `foreach`?

Reply:
Also use QRegularExpression, not QRegExp.